lib/workers/repository/update/pr/code-owners.spec.ts

```diff
@@ -22,13 +23,183 @@ describe('workers/repository/update/pr/code-owners', () => {
       expect(codeOwners).toEqual(['@jimmy']);
     });
 
+    it('respects orphan files', async () => {
+      fs.readLocalFile.mockResolvedValueOnce(
+        ['* @jimmy', 'yarn.lock'].join('\n')
+      );
+      git.getBranchFiles.mockResolvedValueOnce(['yarn.lock']);
+      const codeOwners = await codeOwnersForPr(pr);
+      expect(codeOwners).toEqual([]);
+    });
+
     it('returns more specific code owners', async () => {
       fs.readLocalFile.mockResolvedValueOnce(
         ['* @jimmy', 'package.json @john @maria'].join('\n')
       );
       git.getBranchFiles.mockResolvedValueOnce(['package.json']);
       const codeOwners = await codeOwnersForPr(pr);
-      expect(codeOwners).toEqual(['@john', '@maria']);
+      expect(codeOwners).toEqual(['@john', '@maria', '@jimmy']);
+    });
+
+    describe('returns more specific code owners in monorepos', () => {
+      const mockCodeOwners = codeBlock`
+        # By default, assign to @john
+        #
+        * @john
+
+        # Lockfiles are not owned by anyone, any package dependency update may modify them.
+        # Assigning lockfiles an owner will cause issues as merge requests to be assigned to incorrect users
```

Comment:
```suggestion
        # Assigning an owner to lockfiles causes problems, because pull requests are assigned to wrong users
```

Reply:
Honestly, that shouldn't be an issue anymore, that comment was copied from a CODEOWNERS file used at work, we had to specifically remove any owners from the `yarn.lock` file as renovate would incorrectly assign pull requests. 

With the new implementation, I think it would actually be good to assign owners to lock-files as a "fallback" for cases where a better match has not been found.

Reply:
> Honestly, that shouldn't be an issue anymore, that comment was copied from a `CODEOWNERS `file used at work, we had to specifically remove any owners from the yarn.lock file as renovate would incorrectly assign pull requests.

Whoops, I thought it was a _developer_ comment. It's just a copy/paste of an example `CODEOWNERS `file from a production repository. 🙈 

I'll let you decide what to do with the comment. Though we probably should have a test to confirm we're properly ignoring comments in `CODEOWNERS` files.

> With the new implementation, I think it would actually be good to assign owners to lock-files as a "fallback" for cases where a better match has not been found.

I'll let the maintainers decide. 😉

Reply:
> Though we probably should have a test to confirm we're properly ignoring comments in CODEOWNERS 

Test was already there, you don't see it in the PR diff because I didn't need to modify it.

https://github.com/renovatebot/renovate/blob/11adb16758108c19b7c8887e73d848cb990ecaa7/lib/workers/repository/update/pr/code-owners.spec.ts#L205

Reply:
> > With the new implementation, I think it would actually be good to assign owners to lock-files as a "fallback" for cases where a better match has not been found.

> I'll let the maintainers decide. wink

I'd say that the maintainers here don't have anything to decide, the final configuration of the CODEOWNERS file is up to the maintainers of the target repository, not of the renovate project. If someone wants to add the line `yarn.lock @someone`, then go ahead, in the majority of cases there's going to be a better match (with a higher score) and this line would act as a fallback (for lock file maintenance, for example)

Reply:
i would prefer to not have the extended description here

Reply:
```suggestion
```

@viceice I'm not sure what you mean here:

1. Keep the "old" comment, or
2. Drop the comment completely

If you want option 2 you can apply my suggestion. 😉